pkg/neotest/basic.go

```diff
@@ -0,0 +1,271 @@
+package neotest
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/nspcc-dev/neo-go/pkg/config/netmode"
+	"github.com/nspcc-dev/neo-go/pkg/core/block"
+	"github.com/nspcc-dev/neo-go/pkg/core/blockchainer"
+	"github.com/nspcc-dev/neo-go/pkg/core/fee"
+	"github.com/nspcc-dev/neo-go/pkg/core/native"
+	"github.com/nspcc-dev/neo-go/pkg/core/native/nativenames"
+	"github.com/nspcc-dev/neo-go/pkg/core/state"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/io"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract/callflag"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract/trigger"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm"
+	"github.com/nspcc-dev/neo-go/pkg/vm/emit"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"github.com/stretchr/testify/require"
+)
+
+// Executor is a wrapper over chain state.
+type Executor struct {
+	Chain         blockchainer.Blockchainer
+	Committee     *wallet.Account
+	CommitteeHash util.Uint160
+	Contracts     map[string]*Contract
+}
+
+// NewExecutor creates new executor instance from provided blockchain and committee.
+func NewExecutor(t *testing.T, bc blockchainer.Blockchainer, committee *wallet.Account) *Executor {
+	require.Equal(t, 1, len(bc.GetConfig().StandbyCommittee))
+
+	return &Executor{
+		Chain:         bc,
+		Committee:     committee,
+		CommitteeHash: committee.Contract.ScriptHash(),
+		Contracts:     make(map[string]*Contract),
+	}
+}
+
+// TopBlock returns block with the highest index.
+func (e *Executor) TopBlock(t *testing.T) *block.Block {
+	b, err := e.Chain.GetBlock(e.Chain.GetHeaderHash(int(e.Chain.BlockHeight())))
+	require.NoError(t, err)
+	return b
+}
+
+// NativeHash returns native contract hash by name.
+func (e *Executor) NativeHash(t *testing.T, name string) util.Uint160 {
+	h, err := e.Chain.GetNativeContractScriptHash(name)
+	require.NoError(t, err)
+	return h
+}
+
+// PrepareInvokeNoSign creates new invocation transaction.
+// It doesn't sign the tx.
+func (e *Executor) PrepareInvokeNoSign(t *testing.T, hash util.Uint160, method string, args ...interface{}) *transaction.Transaction {
+	w := io.NewBufBinWriter()
+	emit.AppCall(w.BinWriter, hash, method, callflag.All, args...)
+	require.NoError(t, w.Err)
+
+	script := w.Bytes()
+	tx := transaction.New(script, 0)
+	tx.Nonce = nonce()
+	tx.ValidUntilBlock = e.Chain.BlockHeight() + 1
+	return tx
+}
+
+// PrepareInvoke creates new invocation transaction.
+func (e *Executor) PrepareInvoke(t *testing.T, signer interface{},
+	hash util.Uint160, method string, args ...interface{}) *transaction.Transaction {
+	tx := e.PrepareInvokeNoSign(t, hash, method, args...)
+	return e.SignTx(t, tx, -1, signer)
+}
+
+// SignTx signs a transaction using provided signer.
+// signer can be either *wallet.Account or []*wallet.Account.
```

Comment:
It will clutter code on the caller side, as I expect most of the calls to use single signer. There are only 2 possibilities and simple switch below which ensure nothing unexpected will happen because of this.

Reply:
Fixed this later.